File: Latihan5/C++/soal2.cpp
#include <iostream>
using namespace std;

int main() {
    // Inisialisasi variabel
    double total_penjualan = 0;
    int n;
    
    // Meminta jumlah jenis barang
    cout << "Masukkan jumlah jenis barang: ";
    cin >> n;

    // Loop untuk menginput harga dan jumlah barang
    for (int i = 0; i < n; i++) {
        double harga;
        int jumlah;
        cout << "Masukkan harga barang ke-" << i + 1 << ": ";
        cin >> harga;
        cout << "Masukkan jumlah barang ke-" << i + 1 << ": ";
        cin >> jumlah;
        double subtotal = harga * jumlah;
        total_penjualan += subtotal;
    }

    // Menghitung diskon berdasarkan total penjualan
    double diskon = 0;
    if (total_penjualan > 500000) {
        diskon = 0.1;  // Diskon 10%
    } else if (total_penjualan >= 200000) {
        diskon = 0.05;  // Diskon 5%
    }

    // Menghitung total pembayaran setelah diskon
    double total_pembayaran = total_penjualan - (total_penjualan * diskon);

    // Menampilkan total penjualan dan total pembayaran
    cout << "Total Penjualan: Rp " << total_penjualan << endl;
    cout << "Diskon: " << (diskon * 100) << "%" << endl;
    cout << "Total Pembayaran setelah Diskon: Rp " << total_pembayaran << endl;

    return 0;
}

File: Latihan5/Python/soal2.py
# Inisialisasi variabel
totalPenjualan = 0
n = int(input("Masukkan jumlah jenis barang: "))

# Loop untuk menginput harga dan jumlah barang
for i in range(n):
    harga = int(input(f"Masukkan harga barang ke-{i + 1}: "))
    jumlah = int(input(f"Masukkan jumlah barang ke-{i + 1}: "))
    subTotal = harga * jumlah
    totalPenjualan += subTotal

# Menghitung diskon berdasarkan total penjualan
if totalPenjualan > 500000:
    diskon = 0.1  # Diskon 10%
elif totalPenjualan >= 200000:
    diskon = 0.05  # Diskon 5%
else:
    diskon = 0  # Tidak ada diskon

# Menghitung total pembayaran setelah diskon
totalPembayaran = totalPenjualan - (totalPenjualan * diskon)

# Menampilkan total penjualan dan total pembayaran
print(f"Total Penjualan: Rp {totalPenjualan:.2f}")
print(f"Diskon: {diskon * 100}%")
print(f"Total Pembayaran setelah Diskon: Rp {totalPembayaran:.2f}")
